### PR TITLE
zliu - fix swan bug & enable sse service in go-swan

### PR DESCRIPTION
**Pull request by liuzhengbaihe**

### CODE REVIEW

**Comment by upccup** [comment]
+1

File: go-swan/client.go [hunk 1]
```diff
@@ -32,18 +32,28 @@ type swanClient struct {
 	httpClient *http.Client
 	// a custom logger for debug log messages
 	debugLog *log.Logger
+	hosts    *cluster
```
Comment (cmingxu) [inline go-swan/client.go]: managers ?

File: go-swan/cluster.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+const (
+	memberStatusUp   = 0
+	memberStatusDown = 1
+)
+
+// the status of a member node
+type memberStatus int
+
+// cluster is a collection of swan nodes
+type cluster struct {
```
Comment (cmingxu) [inline go-swan/cluster.go]: SwanCluster

File: go-swan/cluster.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	sync.RWMutex
+	// a collection of nodes
+	members []*member
+	// the http client
+	client *http.Client
+}
+
+// member represents an individual endpoint
+type member struct {
```
Comment (cmingxu) [inline go-swan/cluster.go]: Manager ?

File: go-swan/cluster.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	}
+
+	return &cluster{
+		client:  client,
+		members: members,
+	}, nil
+}
+
+// retrieve the current member, i.e. the current endpoint in use
+func (c *cluster) getMember() (string, error) {
```
Comment (cmingxu) [inline go-swan/cluster.go]: getManager

File: go-swan/cluster.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	memberStatusUp   = 0
```
Comment (cmingxu) [inline go-swan/cluster.go]: MANAGER_REACHABLE

File: go-swan/cluster.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+	memberStatusDown = 1
```
Comment (cmingxu) [inline go-swan/cluster.go]: MANAGER_UNREACHABLE

File: go-swan/cluster.go [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+		if n.status == memberStatusUp && n.endpoint == endpoint {
+			n.status = memberStatusDown
+			go c.healthCheckNode(n)
+			break
+		}
+	}
+}
+
+// healthCheckNode performs a health check on the node and when active updates the status
+func (c *cluster) healthCheckNode(node *member) {
```
Comment (cmingxu) [inline go-swan/cluster.go]: healthCheckManager?

**Comment by cmingxu** [comment]
LGTM

**Pull request merged**